### ISSUE TITLE
Comments before first program unit get lost

### ISSUE THREAD

**Issue by isd-bbierbaum** [lfortran/lfortran#1558]
If this code

! comment
subroutine s
end subroutine s

is treated with `lfortran fmt`

the result is

subroutine s()
end subroutine s

As far as I can see, there is nothing in the AST to represent comments outside of subroutines or functions. In my experience with real-world Fortran code, it is quite common to precede a subroutine with a comment block that describes it, so this would all get lost when formatting the code with LFortran.

I'm using

LFortran version: 0.18.0-1119-gc648a8d82
Platform: Linux
Default target: x86_64-unknown-linux-gnu

**Comment by certik** [comment]
Yes, I think it's the very first comment in the file that is currently lost. Try to put some routine at the top, it should be preserved.

**Comment by BorisBierbaumIsd** [comment]
You're right. Only the comments before the first routine are lost.
